collect_app/src/main/java/org/odk/collect/android/tasks/SaveToDiskTask.java

```diff
@@ -183,28 +197,31 @@ private void updateInstanceDatabase(boolean incomplete, boolean canEditAfterComp
                 values.putAll(geometryContentValues);
             }
 
-            int updated = Collect.getInstance().getContentResolver().update(
-                uri, values, null, null);
+            int updated = Collect.getInstance().getContentResolver().update(uri, values, null, null);
             if (updated > 1) {
                 Timber.w("Updated more than one entry, that's not good: %s", uri.toString());
             } else if (updated == 1) {
                 Timber.i("Instance successfully updated");
             } else {
-                Timber.e("Instance doesn't exist but we have its Uri!! %s", uri.toString());
+                Timber.w("Instance doesn't exist but we have its Uri!! %s", uri.toString());
             }
         } else if (Collect.getInstance().getContentResolver().getType(uri).equals(
                 FormsColumns.CONTENT_ITEM_TYPE)) {
             // If FormEntryActivity was started with a form, then it's likely the first time we're
-            // saving.
-            // However, it could be a not-first time saving if the user has been using the manual
-            // 'save data' option from the menu. So try to update first, then make a new one if that
-            // fails.
+            // saving. However, it could be a not-first time saving if the user has been using the
+            // manual 'save data' option from the menu. So try to update first, then make a new one
+            // if that fails.
             String instancePath = formController.getInstanceFile().getAbsolutePath();
+
             String where = InstanceColumns.INSTANCE_FILE_PATH + "=?";
-            String[] whereArgs = {
-                    instancePath
-            };
-            int updated = new InstancesDao().updateInstance(values, where, whereArgs);
+            int updated = new InstancesDao().updateInstance(values, where, new String[] {instancePath});
+
+            // Set uri to handle encrypted case (see exportData)
+            InstancesRepository instances = new DatabaseInstancesRepository();
+            Instance instance = instances.getByPath(instancePath);
```

Comment:
I decided to introduce `getByPath` here because we'll definitely need it and it allowed me to hide the logic related to cursors and not getting exactly one result back (though I still have to handle the null case here).